vk/options.go

```diff
@@ -10,22 +10,36 @@ import (
 	"github.com/suborbital/vektor/vlog"
 )
 
+// RouterWrapper provides a function signature to implement wrappers for routing. A good use case is to pass in an
+// opentelemetry mux wrapper.
+type RouterWrapper func(handler http.Handler) http.Handler
+
 // Options are the available options for Server
 type Options struct {
-	AppName     string `env:"_APP_NAME"`
-	Domain      string `env:"_DOMAIN"`
-	HTTPPort    int    `env:"_HTTP_PORT"`
-	TLSPort     int    `env:"_TLS_PORT"`
-	TLSConfig   *tls.Config
-	EnvPrefix   string
-	QuietRoutes []string
-	Logger      *vlog.Logger
+	AppName       string `env:"_APP_NAME"`
+	Domain        string `env:"_DOMAIN"`
+	HTTPPort      int    `env:"_HTTP_PORT"`
+	TLSPort       int    `env:"_TLS_PORT"`
+	TLSConfig     *tls.Config
+	EnvPrefix     string
+	QuietRoutes   []string
+	Logger        *vlog.Logger
+	RouterWrapper RouterWrapper
 
 	PreRouterInspector func(http.Request)
 }
 
+// defaultRouterWrapper is a default pass through option for a wrapper. This does not wrap the handler in anything.
+var defaultRouterWrapper = func(innerHandler http.Handler) http.Handler {
+	return innerHandler
+}
+
 func newOptsWithModifiers(mods ...OptionsModifier) *Options {
 	options := &Options{}
+
+	// set the default route wrapper to the noop pass through one.
+	options.RouterWrapper = defaultRouterWrapper
```

Comment:
Normally I'd do this in `opts.finalize`, but not a big deal either way

Reply:
The reason I did it here is because one of the options is about the router wrapper. If I declare the default here, all I have to do is loop through the passed in router wrapper, which will reassign it to what the user wanted, and we're done, saving us an extra if check in finalise.

Somewhat related, I think this should be cleaned up, because the configuration, or the changing of bits of the router / server happens in all of these places:
1. server.New
2. newOptsWithModifiers
3. server.Start
4. server.finalize

Keeping track of what gets set where is somewhat confusing. Ideally everything would be done in the New with their default values, then the option modifiers loop through, and then the only thing Start does is start listening on the configured places